.travis.yml

```diff
@@ -1,214 +1,114 @@
-          sources:
-            - ubuntu-toolchain-r-test
-          packages:
-            - g++-7
-      env:
-        - MATRIX_EVAL="CC=gcc-7 && CXX=g++-7"
-
-    - os: linux
-      addons:
-        apt:
-          sources:
-            - ubuntu-toolchain-r-test
-            - llvm-toolchain-precise-3.6
-          packages:
-            - clang-3.6
-      env:
-        - MATRIX_EVAL="CC=clang-3.6 && CXX=clang++-3.6"
-
-    - os: linux
-      addons:
-        apt:
-          sources:
-            - ubuntu-toolchain-r-test
-            - llvm-toolchain-precise-3.7
-          packages:
-            - clang-3.7
-      env:
-        - MATRIX_EVAL="CC=clang-3.7 && CXX=clang++-3.7"
-
-    - os: linux
-      addons:
+    # Clang 5.0.0
+    - env:
+        - openPMD_USE_MPI=ON openPMD_USE_HDF5=ON openPMD_USE_ADIOS1=OFF openPMD_USE_ADIOS2=OFF
+      compiler: clang
+      addons: &mpi_deps
         apt:
-          sources:
-            - ubuntu-toolchain-r-test
-            - llvm-toolchain-precise-3.8
           packages:
-            - clang-3.8
-      env:
-        - MATRIX_EVAL="CC=clang-3.8 && CXX=clang++-3.8"
-
-    - os: linux
-      addons:
-        apt:
-          sources:
-            - llvm-toolchain-trusty-3.9
-          packages:
-            - clang-3.9
-      env:
-        - MATRIX_EVAL="CC=clang-3.9 && CXX=clang++-3.9"
-
-    - os: linux
-      addons:
-        apt:
-          sources:
-            - llvm-toolchain-trusty-4.0
-          packages:
-            - clang-4.0
-      env:
-        - MATRIX_EVAL="CC=clang-4.0 && CXX=clang++-4.0"
-
-    - os: linux
-      addons:
-        apt:
-          sources:
-            - llvm-toolchain-trusty-5.0
-          packages:
-            - clang-5.0
-      env:
-        - MATRIX_EVAL="CC=clang-5.0 && CXX=clang++-5.0"
-
-#env:
-#  matrix:
-#    - PARALLEL=ON
-#    - PARALLEL=OFF
-
-env:
- global:
-  - CMAKE_VERSION=3.10
-  - CMAKE_PATCH=2
-  - BOOST_MINOR=64
-  - BOOST_PATCH=0
-  - HDF5_MINOR=10
-  - HDF5_PATCH=1
-  - ADIOS1_MINOR=13
-  - ADIOS1_PATCH=0
-  - ADIOS2_MINOR=1
-  - ADIOS2_PATCH=0
- #matrix:
-  #- BOOST_MINOR=58 BOOST_PATCH=0
-  #- BOOST_MINOR=64 BOOST_PATCH=0
-  #- BOOST_MINOR=65 BOOST_PATCH=1
-  #- BOOST_MINOR=66 BOOST_PATCH=0
-
-before_install:
- - eval "${MATRIX_EVAL}"
-
- - sudo sh scripts/travis/cache_ci_dependencies.sh
-
- - cd .cache
- - cd CMake/v${CMAKE_VERSION}.${CMAKE_PATCH}
- - sudo make install -j4
- - cd ..
-
- - cd boost_1_${BOOST_MINOR}_${BOOST_PATCH}
- - sudo ./b2 install -d0 -j4
- - cd ..
-
- - cd hdf5-1.${HDF5_MINOR}.${HDF5_PATCH}
- - sudo make install -j4
- - cd ..
-
- - cd adios-1.${ADIOS1_MINOR}.${ADIOS1_PATCH}
- - sudo make install -j4
- - cd ..
-
- - cd ADIOS2/v2.${ADIOS2_MINOR}.${ADIOS2_PATCH}
- - sudo make install -j4
- - cd ..
+            - *common_deps
+            - libopenmpi-dev
+            - openmpi-bin
+    - env:
+        - openPMD_USE_MPI=OFF openPMD_USE_HDF5=ON openPMD_USE_ADIOS1=OFF openPMD_USE_ADIOS2=OFF
+      compiler: clang
+    # GCC 4.8.5
```

Comment:
if gcc 4.8 does not cut it we just go for gcc 4.9+: https://docs.travis-ci.com/user/languages/c/